src/main/scala/com/microsoft/hyperspace/index/sources/default/DefaultFileBasedSource.scala

```diff
@@ -67,8 +67,19 @@ class DefaultFileBasedSource(private val spark: SparkSession) extends FileBasedS
         val sourceDataProperties =
           Hdfs.Properties(Content.fromLeafFiles(files, fileIdTracker).get)
         val fileFormatName = fileFormat.asInstanceOf[DataSourceRegister].shortName
+
+        // Store basePath of hive-partitioned data sources, if applicable.
+        val basePath = PathUtils.extractBasePath(location.partitionSpec())
+
         // "path" key in options can incur multiple data read unexpectedly.
-        val opts = options - "path"
+        // Since "options" is case-insensitive map, it will change any previous entries of
+        // "basePath" to lowercase "basepath", making it unusable.
+        // Remove lowercase "basepath" and add proper cased "basePath".
+        val opts = basePath match {
+          case Some(path) => Map("basePath" -> path) ++ options - "path" - "basepath"
+          case _ => options - "path" - "basepath"
+        }
```

Comment:
```suggestion
        // Since "options" is case-insensitive map, it will change any previous entries of
        // "basePath" to lowercase "basepath", making it unusable.
        // Remove lowercase "basepath" and add proper cased "basePath".
        val opts =  options - "path" - "basepath" + (basePath match {
          case Some(path) => "basePath" -> path
          case _ => Nil
        }
```
Could you check this? suggestion is not tested.

Reply:
no this won't work because `options` is case insensitive. opts will change "basePath" to "basepath" making it unusable